src/OpenLoco/src/Objects/IndustryObject.cpp

```diff
@@ -171,26 +173,162 @@ namespace OpenLoco
     // 0x00458CD9
     void IndustryObject::load(const LoadedObjectHandle& handle, stdx::span<const std::byte> data, ObjectManager::DependentObjects* dependencies)
     {
-        Interop::registers regs;
-        regs.esi = Interop::X86Pointer(this);
-        regs.ebx = handle.id;
-        regs.ecx = enumValue(handle.type);
-        Interop::call(0x00458CD9, regs);
-        if (dependencies != nullptr)
-        {
-            auto* depObjs = addr<0x0050D158, uint8_t*>();
-            dependencies->required.resize(*depObjs++);
-            if (!dependencies->required.empty())
+        auto remainingData = data.subspan(sizeof(IndustryObject));
+
+        {
+            auto strRes = ObjectManager::loadStringTable(remainingData, handle, 0);
+            name = strRes.str;
+            remainingData = remainingData.subspan(strRes.tableLength);
+            strRes = ObjectManager::loadStringTable(remainingData, handle, 1);
+            var_02 = strRes.str;
+            remainingData = remainingData.subspan(strRes.tableLength);
+            strRes = ObjectManager::loadStringTable(remainingData, handle, 2);
+            // Unused??
+            remainingData = remainingData.subspan(strRes.tableLength);
+            strRes = ObjectManager::loadStringTable(remainingData, handle, 3);
+            nameClosingDown = strRes.str;
+            remainingData = remainingData.subspan(strRes.tableLength);
+            strRes = ObjectManager::loadStringTable(remainingData, handle, 4);
+            nameUpProduction = strRes.str;
+            remainingData = remainingData.subspan(strRes.tableLength);
+            strRes = ObjectManager::loadStringTable(remainingData, handle, 5);
+            nameDownProduction = strRes.str;
+            remainingData = remainingData.subspan(strRes.tableLength);
+            strRes = ObjectManager::loadStringTable(remainingData, handle, 6);
+            nameSingular = strRes.str;
+            remainingData = remainingData.subspan(strRes.tableLength);
+            strRes = ObjectManager::loadStringTable(remainingData, handle, 7);
+            namePlural = strRes.str;
+            remainingData = remainingData.subspan(strRes.tableLength);
+        }
+
+        buildingPartHeight = reinterpret_cast<const uint8_t*>(remainingData.data());
```

Comment:
In this function, it's not obvious at first glance to see where what is loaded. I wonder how best to improve this with little effort.

Perhaps just add a few visual cues / comments before each major block? I suppose it would be even nicer to break things up into smaller (private) functions, but I'm not sure that would be worth doing.